src/script/auth/module/action/ClientAction.ts

```diff
@@ -62,20 +62,21 @@ export class ClientAction {
     entropyData?: Uint8Array,
   ): ThunkAction => {
     return async (dispatch, getState, {core, actions: {clientAction, webSocketAction}}) => {
-      dispatch(ClientActionCreator.startInitializeClient());
-      try {
-        const creationStatus = await core.initClient(
-          {clientType, password, verificationCode},
-          clientAction.generateClientPayload(clientType),
-          entropyData,
-        );
-        await dispatch(clientAction.doGetAllClients());
-        await dispatch(webSocketAction.listen());
-        dispatch(ClientActionCreator.successfulInitializeClient(creationStatus));
-      } catch (error) {
-        dispatch(ClientActionCreator.failedInitializeClient(error));
-        throw error;
-      }
+      const localClient = await core.initClient();
+      const creationStatus = localClient
+        ? {isNew: false, client: localClient}
+        : {
+            isNew: true,
+            client: await core.registerClient(
```

Comment:
what happens if registerClient throws an error?

Reply:
We are going to have a nasty error in the console, and that's fine there are not exception that we should handle at this point (it would mean that the backend is having hard time I guess)

Reply:
Actually, let me try the "to many devices" scenario now 🤔 
You might have a point

Reply:
Oh, it's handled correctly as well. We just don't have the `failedInitializeClient` call (that was useless anyway)
But we still throw the backend error

Reply:
Yep, confirmed with a test 👌 
Thanks for pointing it out